coderd/templateversions_test.go

```diff
@@ -448,6 +450,207 @@ func TestPatchActiveTemplateVersion(t *testing.T) {
 	})
 }
 
+func TestTemplateVersionPlan(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK", func(t *testing.T) {
+		t.Parallel()
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		resource := &proto.Resource{
+			Name: "cool-resource",
+			Type: "cool_resource_type",
+		}
+
+		client := coderdtest.New(t, &coderdtest.Options{APIRateLimit: -1, IncludeProvisionerD: true})
+		user := coderdtest.CreateFirstUser(t, client)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+			Parse: echo.ParseComplete,
+			Provision: []*proto.Provision_Response{
+				{
+					Type: &proto.Provision_Response_Log{
+						Log: &proto.Log{},
+					},
+				},
+				{
+					Type: &proto.Provision_Response_Complete{
+						Complete: &proto.Provision_Complete{
+							Resources: []*proto.Resource{resource},
+						},
+					},
+				},
+			},
+		})
+		_ = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+
+		// Create template version plan
+		after := time.Now()
+		job, err := client.CreateTemplateVersionPlan(ctx, version.ID, codersdk.CreateTemplateVersionPlanRequest{
+			ParameterValues: []codersdk.CreateParameterRequest{},
+		})
+		require.NoError(t, err)
+
+		// Fetch template version plan
+		newJob, err := client.TemplateVersionPlan(ctx, version.ID, job.ID)
+		require.NoError(t, err)
+		require.Equal(t, job, newJob)
+
+		// Stream logs
+		logs, err := client.TemplateVersionPlanLogsAfter(ctx, version.ID, job.ID, after)
+		require.NoError(t, err)
+
+		logsDone := make(chan struct{})
+		go func() {
+			defer close(logsDone)
+
+			logCount := 0
+			for range logs {
+				logCount++
+			}
+			assert.GreaterOrEqual(t, logCount, 1, "unexpected log count")
+		}()
+
+		// Wait for the job to complete
+		require.Eventually(t, func() bool {
+			job, err := client.TemplateVersionPlan(ctx, version.ID, job.ID)
+			assert.NoError(t, err)
+
+			return job.Status == codersdk.ProvisionerJobSucceeded
```

Comment:
(non-blocking): cool thing I found out about `testify/assert`: you can `return assert.Equal(t, codersdk.ProvisionerJobSucceeded, job.Status)` and it will be treated as a bool.

Reply:
would that still work with the require.Eventually?

Reply:
Seems to work fine for me

Reply:
If you do it in this case then it will fail the job even though it's supposed to be an Eventually right? It doesn't work for me when I try it here

Reply:
Ah I forgot it still calls `t.Errorf`, please disregard my spurious advice